src/Dhall.hs

```diff
@@ -1333,3 +1340,80 @@ field key valueType =
           )
           ( Data.Functor.Compose.Compose extractBody )
       )
+
+{-| The 'InputRecordType' divisible (contravariant) functor allows you to build 
+    an 'InputType' injector for a Dhall record.
+
+    For example, let's take the following Haskell data type:
+
+> data Project = Project
+>   { projectName :: Text
+>   , projectDescription :: Text
+>   , projectStars :: Natural
+>   }
+
+    And assume that we have the following Dhall record that we would like to
+    parse as a @Project@:
+
+> { name =
+>     "dhall-haskell"
+> , description =
+>     "A configuration language guaranteed to terminate"
+> , stars =
+>     289
+> }
+
+    Our injector has type 'InputType' @Project@, but we can't build that out of any
+    smaller injectors, as 'InputType's cannot be combined (they are only 'Contravariant's).
+    However, we can use an 'InputRecordType' to build an 'InputType' for @Project@:
+
+> injectProject :: InputType Project
+> injectProject =
+>   inputRecord
+>     (  adapt >$< inputField "name" (inject @Text)
+>              >*< inputField "description" (inject @Text)
+>              >*< inputField "stars" (inject @Natural)
+>     )
+>   where
+>     adapt (Project{..}) = (projectName, (projectDescription, projectStars))
+
+-}
+
+-- | Infix 'contramap'
+(>$<) :: Contravariant f => (a -> b) -> f b -> f a
```

Comment:
I just realized that the `>$<` operator already exists in the `contravariant` package:

http://hackage.haskell.org/package/contravariant-1.5/docs/Data-Functor-Contravariant.html

... so we can reuse that

Reply:
I can drop that definitiion and just rexport the contravariant definition. Give me a few minutes.